### ISSUE TITLE
Improvement Suggestion: Related Videos in YouTube Embed

### ISSUE THREAD

**Issue by eduardokraus** [sokunthearithmakara/mod_interactivevideo#9]
Currently, when using YouTube embeds in "interactions", related videos are still being displayed, which can cause distractions or show inappropriate content for the context.

I’d like to suggest a fix by using YouTube's enhanced privacy mode. Please note that the `rel=0` parameter no longer prevents related videos from being displayed if the user is logged into a Google account (YouTube, Gmail, etc.).

The `&rel=0` parameter still works correctly only for users who are not logged in. To bypass this limitation, I recommend using the enhanced privacy mode URL:

```
https://www.youtube-nocookie.com/embed/[id]?rel=0
```

This change should ensure that related videos are not shown regardless of the user's login status.

![Captura de Tela 2024-10-11 às 16 33 14](https://github.com/user-attachments/assets/0a3f7875-7d9d-4da6-872d-fb51a5504392)

**Comment by sokunthearithmakara** [comment]
@eduardokraus I used the nocookie version in the earlier version of the plugin and found that the related video visibility is inconsistent regardless of the host. I understand that can be distracting for learners, so I added a support for teacher to choose between regular host (youtube.com) or nocookie depending on the address of the video.

**Comment by eduardokraus** [comment]
Thanks

**Comment by sokunthearithmakara** [comment]
@eduardokraus I did further research and found that if the YouTube video iframe is small (probably less than 400px) or the iframe is in an invisible parent element, the related videos are not shown. So the trick is to either hide the video until the video finishes loading or make the iframe small first and resize it when the player is ready. I tested it out and it worked perfectly. I hope this info is helpful in case you want to implement it in your plugins.
